api/v1alpha1/openstackimage_types.go

```diff
@@ -63,16 +51,21 @@ type OpenStackImageSpec struct {
 	// image.
 	MinRAM int `json:"minRam,omitempty"`
 
+	// Name of the OpenStack resource.
+	Name string `json:"name,omitempty"`
+
 	// protected is whether the image is not deletable.
-	Protected *bool `json:"protected,omitempty"`
+	// +kubebuilder:default:=false
```

Comment:
So the way I think we should go about optional fields is: we don't pass them. OpenStack (in this case, Glance) will use its default value; let's not assume it's always going to be "false". I'd remove the `kubebuilder:default` annotation.